### PR TITLE
Build docker image

### PR DESCRIPTION
**Pull request by michalwski**
This PR introduces a Dockerfile and script for build docker image.

### CODE REVIEW

File: tools/wait_for_healthcheck.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# Waiting for a running container to become healthy.
+# Waits if no container was found with reason "no such object".
+# Waits if the container does not healthcheck configured.
```
Comment (sebb7) [inline tools/wait_for_healthcheck.sh]: Typo here?

File: build_docker_image.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Get current repo version
+VERSION=$(git rev-parse --short HEAD)
+ERLANG_VSN=${OTP_VSN:-21.3.8.7-1}
+echo "ERLANG/OTP ${ERLANG_VSN}"
+
+docker build \
+	-f Dockerfile \
```
[the remark this answers is not in the document]
Comment (michalwski) [reply build_docker_image.sh]: I think it doesn't hurt to be specific about the `Dockerfile` to use. I'd leave it as is.

File: build_docker_image.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+	-t amoc-arsenal-xmpp:latest \
```
[the remark this answers is not in the document]
Comment (michalwski) [reply build_docker_image.sh]: This script is mainly to test the image can be build and then container can be started. If you want you can improve the script as you described.

File: build_docker_image.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Get current repo version
+VERSION=$(git rev-parse --short HEAD)
+ERLANG_VSN=${OTP_VSN:-21.3.8.7-1}
```
[the remark this answers is not in the document]
Comment (michalwski) [reply build_docker_image.sh]: Yes, we need that. The docker image can be build without the `build_docker_image` script.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
+FROM phusion/baseimage AS builder
+MAINTAINER Erlang Solutions <mongoose-im@erlang-solutions.com>
+
+RUN useradd -ms /bin/bash amoc
```
[the remark this answers is not in the document]
Comment (michalwski) [reply Dockerfile]: I think we don't need the user here, only in the final image.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+RUN cd amoc_arsenal_build && \
+    ./rebar3 release
+
+FROM phusion/baseimage
+MAINTAINER Erlang Solutions <mongoose-im@erlang-solutions.com>
+
+ARG vsn
+ARG amoc_arsenal_home=/home/amoc/amoc_arsenal_xmpp
+ARG amoc_github=https://raw.githubusercontent.com/esl/amoc
```
[the remark this answers is not in the document]
Comment (michalwski) [reply Dockerfile]: Probably not anymore, let's remove it.

File: tools/wait_for_healthcheck.sh (hunk 1 above)
Comment (michalwski) [reply tools/wait_for_healthcheck.sh]: You are probably right :)

File: build_docker_image.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebb7) [reply build_docker_image.sh]: Would not this image persist on developer host while running load tests with usage of clt-swarm? When two load tests for two different scenarios will be run one image would lost name.

[the remark this answers is not in the document]
Comment (sebb7) [reply build_docker_image.sh]: @michalwski you are right! Sorry for that :) Clt-swarm manages images versions.

**Pull request merged**